TestPeople is really, really broken

Just doing the excerpts for the failing tests, because this test suite has a lot of individual tests.  Also...there are a lot more failures.  This one is very, very broken.  You don't even have to really try to make this fail.  But what's worse is that it appears that some of these failures are not actually marking tests as `FAIL`ed, so it wouldn't show up in a CI run's report as anything more than logged messages.

See #237 

```
=== RUN   TestPeople/negative_hops/badger
    people_test.go:62: created(38) alice as @7jHOZszoMSOyB9maZORSYnzOIXrWSJaZDJqfwG1OU/c=.ed25519 (algo:ed25519)
    people_test.go:62: created(39) bob as @xfKhHxHPYqlMeJ0aWznFANeUte7QHQ1znkwEEPE91e0=.ed25519 (algo:ed25519)
    people_test.go:265:
                Error Trace:    /home/go-ssb/go-ssb/graph/people_test.go:265
                Error:          Should be true
                Test:           TestPeople/negative_hops/badger
                Messages:       did not find peer alice in graph lookup table (len:0)
    people_test.go:265:
                Error Trace:    /home/go-ssb/go-ssb/graph/people_test.go:265
                Error:          Should be true
                Test:           TestPeople/negative_hops/badger
                Messages:       did not find peer bob in graph lookup table (len:0)
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x89544b]

goroutine 1041 [running]:
github.com/dgraph-io/badger/v3.(*memTable).IncrRef(...)
        /home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.5/memtable.go:231
github.com/dgraph-io/badger/v3.(*DB).getMemTables(0xc00dee6480)
        /home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.5/db.go:696 +0x12b
github.com/dgraph-io/badger/v3.(*DB).get(0xc00dee6480, {0xc00ded2000, 0x27, 0x27})
        /home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.5/db.go:730 +0x9d
github.com/dgraph-io/badger/v3.(*Txn).Get(0xc0022f4080, {0xc000024100, 0x1f, 0x20})
        /home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.5/txn.go:478 +0x2a7
github.com/ssbc/margaret/indexes/badger.(*index).GetSeq.func1(0xc00dee6480?)
        /home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/indexes/badger/index.go:387 +0x32
github.com/dgraph-io/badger/v3.(*DB).View(0xc00dece1b8?, 0xc0022aee10)
        /home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.5/txn.go:806 +0x9c
github.com/ssbc/margaret/indexes/badger.(*index).GetSeq(0xc0024b4140)
        /home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/indexes/badger/index.go:386 +0x198
github.com/ssbc/margaret/indexes.(*sinkIndex).QuerySpec(0x0?)
        /home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/indexes/sinkindex.go:30 +0x25
github.com/ssbc/go-ssb/graph.serveLog.func1()
        /home/go-ssb/go-ssb/graph/builder_test.go:330 +0xdd
created by github.com/ssbc/go-ssb/graph.serveLog
        /home/go-ssb/go-ssb/graph/builder_test.go:327 +0x165
FAIL    github.com/ssbc/go-ssb/graph    1.648s
FAIL
```

```
=== RUN   TestPeople/unfollow/badger
    people_test.go:62: created(3) alice as @mTABbvnx4tu2VAm3SpLCBOcyeklDySXdaQA3wMlRomw=.ed25519 (algo:ed25519)
    people_test.go:62: created(4) bob as @rNM8Fp8TcXo7XXoFE3uFO/B7c+QdN+JdcPEc45VQlaI=.ed25519 (algo:ed25519)
    people_test.go:62: created(5) claire as @WGW2y4cA6nsp3h5syPJJIX0WwJjWhsngfFpVuY78YG8=.ed25519 (algo:ed25519)
    people_test.go:265:
                Error Trace:    /home/go-ssb/go-ssb/graph/people_test.go:265
                Error:          Should be true
                Test:           TestPeople/unfollow/badger
                Messages:       did not find peer alice in graph lookup table (len:0)
    people_test.go:265:
                Error Trace:    /home/go-ssb/go-ssb/graph/people_test.go:265
                Error:          Should be true
                Test:           TestPeople/unfollow/badger
                Messages:       did not find peer bob in graph lookup table (len:0)
    people_test.go:265:
                Error Trace:    /home/go-ssb/go-ssb/graph/people_test.go:265
                Error:          Should be true
                Test:           TestPeople/unfollow/badger
                Messages:       did not find peer claire in graph lookup table (len:0)
level=warn t=24.605214ms msg="authbypass - trust on first use"
    people_test.go:274:
                Error Trace:    /home/go-ssb/go-ssb/graph/people_test.go:274
                Error:          Received unexpected error:
                                auth assert: host(alice) accepted remote(bob) (dist:0)
                Test:           TestPeople/unfollow/badger
                Messages:       assertion #3 failed
    people_test.go:278: warning: failed to dump graph to SVG RenderSVG: dot command failed: exec: "dot": executable file not found in $PATH
level=warn t=24.962874ms msg="authbypass - trust on first use"
    people_test.go:274:
                Error Trace:    /home/go-ssb/go-ssb/graph/people_test.go:274
                Error:          Received unexpected error:
                                follows assert failed - wanted true
                Test:           TestPeople/unfollow/badger
                Messages:       assertion #5 failed
    people_test.go:278: warning: failed to dump graph to SVG RenderSVG: dot command failed: exec: "dot": executable file not found in $PATH
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x89544b]

goroutine 129 [running]:
github.com/dgraph-io/badger/v3.(*memTable).IncrRef(...)
        /home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.5/memtable.go:231
github.com/dgraph-io/badger/v3.(*DB).getMemTables(0xc0000f7b00)
        /home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.5/db.go:696 +0x12b
github.com/dgraph-io/badger/v3.(*DB).get(0xc0000f7b00, {0xc0187a74d0, 0x27, 0x27})
        /home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.5/db.go:730 +0x9d
github.com/dgraph-io/badger/v3.(*Txn).Get(0xc018753080, {0xc000025380, 0x1f, 0x20})
        /home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.5/txn.go:478 +0x2a7
github.com/ssbc/margaret/indexes/badger.(*index).GetSeq.func1(0xc0000f7b00?)
        /home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/indexes/badger/index.go:387 +0x32
github.com/dgraph-io/badger/v3.(*DB).View(0xc00de68cd8?, 0xc0022f0e10)
        /home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.5/txn.go:806 +0x9c
github.com/ssbc/margaret/indexes/badger.(*index).GetSeq(0xc00cda8320)
        /home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/indexes/badger/index.go:386 +0x198
github.com/ssbc/margaret/indexes.(*sinkIndex).QuerySpec(0x0?)
        /home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/indexes/sinkindex.go:30 +0x25
github.com/ssbc/go-ssb/graph.serveLog.func1()
        /home/go-ssb/go-ssb/graph/builder_test.go:330 +0xdd
created by github.com/ssbc/go-ssb/graph.serveLog
        /home/go-ssb/go-ssb/graph/builder_test.go:327 +0x165
FAIL    github.com/ssbc/go-ssb/graph    0.104s
FAIL
```

```
=== RUN   TestPeople/friends2/badger
    people_test.go:62: created(11) alice as @QgppJkKXvISpSREEymrm6Eo8ueXdCEsT4NXLblMFNbQ=.ed25519 (algo:ed25519)
    people_test.go:62: created(12) bob as @M/i2gniJ9pXbZUVLzZlJcVTF35YuHdEazGykNWFcioc=.ed25519 (algo:ed25519)
    people_test.go:62: created(13) claire as @Rxiw1qV63OL3ltw5UoBAeNLS2mYuBPNvjRt08nf+DzM=.ed25519 (algo:ed25519)
    people_test.go:62: created(14) debora as @eKiXUgaRPSdK14Ho2tWGtz88yZrhJTgGwZZohgiYPyE=.ed25519 (algo:ed25519)
    people_test.go:265:
                Error Trace:    /home/go-ssb/go-ssb/graph/people_test.go:265
                Error:          Should be true
                Test:           TestPeople/friends2/badger
                Messages:       did not find peer alice in graph lookup table (len:0)
    people_test.go:265:
                Error Trace:    /home/go-ssb/go-ssb/graph/people_test.go:265
                Error:          Should be true
                Test:           TestPeople/friends2/badger
                Messages:       did not find peer bob in graph lookup table (len:0)
    people_test.go:265:
                Error Trace:    /home/go-ssb/go-ssb/graph/people_test.go:265
                Error:          Should be true
                Test:           TestPeople/friends2/badger
                Messages:       did not find peer claire in graph lookup table (len:0)
    people_test.go:265:
                Error Trace:    /home/go-ssb/go-ssb/graph/people_test.go:265
                Error:          Should be true
                Test:           TestPeople/friends2/badger
                Messages:       did not find peer debora in graph lookup table (len:0)
    people_test.go:274:
                Error Trace:    /home/go-ssb/go-ssb/graph/people_test.go:274
                Error:          Received unexpected error:
                                follows assert failed - wanted true
                Test:           TestPeople/friends2/badger
                Messages:       assertion #1 failed
    people_test.go:278: warning: failed to dump graph to SVG RenderSVG: dot command failed: exec: "dot": executable file not found in $PATH
    people_test.go:274:
                Error Trace:    /home/go-ssb/go-ssb/graph/people_test.go:274
                Error:          Received unexpected error:
                                follows assert failed - wanted true
                Test:           TestPeople/friends2/badger
                Messages:       assertion #2 failed
    people_test.go:278: warning: failed to dump graph to SVG RenderSVG: dot command failed: exec: "dot": executable file not found in $PATH
    people_test.go:274:
                Error Trace:    /home/go-ssb/go-ssb/graph/people_test.go:274
                Error:          Received unexpected error:
                                follows assert failed - wanted true
                Test:           TestPeople/friends2/badger
                Messages:       assertion #3 failed
    people_test.go:278: warning: failed to dump graph to SVG RenderSVG: dot command failed: exec: "dot": executable file not found in $PATH
    people_test.go:274:
                Error Trace:    /home/go-ssb/go-ssb/graph/people_test.go:274
                Error:          Received unexpected error:
                                dist: make dijkstra failed: ssb/graph: no such from: @QgppJkKXvISpSREEymrm6Eo8ueXdCEsT4NXLblMFNbQ=.ed25519
                Test:           TestPeople/friends2/badger
                Messages:       assertion #4 failed
    people_test.go:278: warning: failed to dump graph to SVG RenderSVG: dot command failed: exec: "dot": executable file not found in $PATH
level=warn t=140.690909ms msg="authbypass - trust on first use"
    people_test.go:274:
                Error Trace:    /home/go-ssb/go-ssb/graph/people_test.go:274
                Error:          Received unexpected error:
                                auth assert: host(alice) accepted remote(debora) (dist:0)
                Test:           TestPeople/friends2/badger
                Messages:       assertion #5 failed
    people_test.go:278: warning: failed to dump graph to SVG RenderSVG: dot command failed: exec: "dot": executable file not found in $PATH
level=warn t=141.095554ms msg="authbypass - trust on first use"
    people_test.go:274:
                Error Trace:    /home/go-ssb/go-ssb/graph/people_test.go:274
                Error:          Received unexpected error:
                                auth assert: host(alice) accepted remote(debora) (dist:1)
                Test:           TestPeople/friends2/badger
                Messages:       assertion #6 failed
    people_test.go:278: warning: failed to dump graph to SVG RenderSVG: dot command failed: exec: "dot": executable file not found in $PATH
level=warn t=141.526452ms msg="authbypass - trust on first use"
    builder_test.go:131: closed scenary
```

```
=== RUN   TestPeople/blocks/badger
    people_test.go:62: created(15) alice as @KUM9tEHEDsySlDkbujU1NzGccf+x926aKgZvwCC5HyE=.ed25519 (algo:ed25519)
    people_test.go:62: created(16) bob as @029ZcpU14U1WceWigylZE8vP/sazYiaozZcAt0Ymoo8=.ed25519 (algo:ed25519)
    people_test.go:265:
                Error Trace:    /home/go-ssb/go-ssb/graph/people_test.go:265
                Error:          Should be true
                Test:           TestPeople/blocks/badger
                Messages:       did not find peer alice in graph lookup table (len:0)
    people_test.go:265:
                Error Trace:    /home/go-ssb/go-ssb/graph/people_test.go:265
                Error:          Should be true
                Test:           TestPeople/blocks/badger
                Messages:       did not find peer bob in graph lookup table (len:0)
    people_test.go:274:
                Error Trace:    /home/go-ssb/go-ssb/graph/people_test.go:274
                Error:          Received unexpected error:
                                Blocks() assert failed - wanted true
                Test:           TestPeople/blocks/badger
                Messages:       assertion #2 failed
    people_test.go:278: warning: failed to dump graph to SVG RenderSVG: dot command failed: exec: "dot": executable file not found in $PATH
    builder_test.go:131: closed scenary
```

```
=== RUN   TestPeople/blocks/badger
    people_test.go:62: created(15) alice as @KUM9tEHEDsySlDkbujU1NzGccf+x926aKgZvwCC5HyE=.ed25519 (algo:ed25519)
    people_test.go:62: created(16) bob as @029ZcpU14U1WceWigylZE8vP/sazYiaozZcAt0Ymoo8=.ed25519 (algo:ed25519)
    people_test.go:265:
                Error Trace:    /home/go-ssb/go-ssb/graph/people_test.go:265
                Error:          Should be true
                Test:           TestPeople/blocks/badger
                Messages:       did not find peer alice in graph lookup table (len:0)
    people_test.go:265:
                Error Trace:    /home/go-ssb/go-ssb/graph/people_test.go:265
                Error:          Should be true
                Test:           TestPeople/blocks/badger
                Messages:       did not find peer bob in graph lookup table (len:0)
    people_test.go:274:
                Error Trace:    /home/go-ssb/go-ssb/graph/people_test.go:274
                Error:          Received unexpected error:
                                Blocks() assert failed - wanted true
                Test:           TestPeople/blocks/badger
                Messages:       assertion #2 failed
    people_test.go:278: warning: failed to dump graph to SVG RenderSVG: dot command failed: exec: "dot": executable file not found in $PATH
    builder_test.go:131: closed scenary
```

```
=== RUN   TestPeople/negative_hops/badger
    people_test.go:62: created(38) alice as @7jHOZszoMSOyB9maZORSYnzOIXrWSJaZDJqfwG1OU/c=.ed25519 (algo:ed25519)
    people_test.go:62: created(39) bob as @xfKhHxHPYqlMeJ0aWznFANeUte7QHQ1znkwEEPE91e0=.ed25519 (algo:ed25519)
    people_test.go:265:
                Error Trace:    /home/go-ssb/go-ssb/graph/people_test.go:265
                Error:          Should be true
                Test:           TestPeople/negative_hops/badger
                Messages:       did not find peer alice in graph lookup table (len:0)
    people_test.go:265:
                Error Trace:    /home/go-ssb/go-ssb/graph/people_test.go:265
                Error:          Should be true
                Test:           TestPeople/negative_hops/badger
                Messages:       did not find peer bob in graph lookup table (len:0)
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x89544b]

goroutine 1044 [running]:
github.com/dgraph-io/badger/v3.(*memTable).IncrRef(...)
        /home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.5/memtable.go:231
github.com/dgraph-io/badger/v3.(*DB).getMemTables(0xc0048a2480)
        /home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.5/db.go:696 +0x12b
github.com/dgraph-io/badger/v3.(*DB).get(0xc0048a2480, {0xc002ba3260, 0x27, 0x27})
        /home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.5/db.go:730 +0x9d
github.com/dgraph-io/badger/v3.(*Txn).Get(0xc00027aa80, {0xc000024640, 0x1f, 0x20})
        /home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.5/txn.go:478 +0x2a7
github.com/ssbc/margaret/indexes/badger.(*index).GetSeq.func1(0xc0048a2480?)
        /home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/indexes/badger/index.go:387 +0x32
github.com/dgraph-io/badger/v3.(*DB).View(0xc00303c008?, 0xc000052e10)
        /home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.5/txn.go:806 +0x9c
github.com/ssbc/margaret/indexes/badger.(*index).GetSeq(2022/12/23 01:42:17 [margaret/indexes/badger] overwrote batch limit 0
```

```
=== RUN   TestPeople/hops_2/badger
    people_test.go:62: created(50) alice as @lfWnBH2njS/EDO6gQ+/182Xk4h/GStHpCNRXP5y2mZE=.ed25519 (algo:ed25519)
    people_test.go:62: created(51) bob as @P/VGLvjY+edi4/zbtMEzq7nw64hWfCPQdwbfiFhGt/g=.ed25519 (algo:ed25519)
    people_test.go:62: created(52) claire as @q1JrC8V4q5iRmUW0N2mryqRkzPdKhe0Cs1rUSGTORfI=.ed25519 (algo:ed25519)
    people_test.go:62: created(53) bobf1 as @LuJzhWvs47r/j+wv611wgH3gBtyCoJWVRfFNqGFi6K8=.ed25519 (algo:ed25519)
    people_test.go:62: created(54) bobf2 as @5iFR8+Pifnya6epCY2aRRoIT+5RdmbYWF2JsJ8lyYaU=.ed25519 (algo:ed25519)
    people_test.go:62: created(55) bobfam1 as @pgm1G9v8fvH/nZhil5hzBevBEFQfr3u52TmU0iGJnf4=.ed25519 (algo:ed25519)
    people_test.go:62: created(56) bobfam2 as @l3AxD6A6pBRhdS9742bSq0R59yJsBn9xyml5wqDg94I=.ed25519 (algo:ed25519)
    people_test.go:62: created(57) bobfam3 as @O+qANDVqY/k32q7P8cGvod140D3AcazTTxRzpF46rsA=.ed25519 (algo:ed25519)
    people_test.go:265:
                Error Trace:    /home/go-ssb/go-ssb/graph/people_test.go:265
                Error:          Should be true
                Test:           TestPeople/hops_2/badger
                Messages:       did not find peer claire in graph lookup table (len:7)
    hops_test.go:125:
                Error Trace:    /home/go-ssb/go-ssb/graph/hops_test.go:125
                                                        /home/go-ssb/go-ss2022/12/23 01:42:36 [margaret/indexes/badger] overwrote batch limit 0
```

```
=== RUN   TestPeople/unfollow/badger
    people_test.go:62: created(3) alice as @mTABbvnx4tu2VAm3SpLCBOcyeklDySXdaQA3wMlRomw=.ed25519 (algo:ed25519)
    people_test.go:62: created(4) bob as @rNM8Fp8TcXo7XXoFE3uFO/B7c+QdN+JdcPEc45VQlaI=.ed25519 (algo:ed25519)
    people_test.go:62: created(5) claire as @WGW2y4cA6nsp3h5syPJJIX0WwJjWhsngfFpVuY78YG8=.ed25519 (algo:ed25519)
    people_test.go:265:
                Error Trace:    /home/go-ssb/go-ssb/graph/people_test.go:265
                Error:          Should be true
                Test:           TestPeople/unfollow/badger
                Messages:       did not find peer bob in graph lookup table (len:0)
    people_test.go:265:
                Error Trace:    /home/go-ssb/go-ssb/graph/people_test.go:265
                Error:          Should be true
                Test:           TestPeople/unfollow/badger
                Messages:       did not find peer claire in graph lookup table (len:0)
    people_test.go:265:
                Error Trace:    /home/go-ssb/go-ssb/graph/people_test.go:265
                Error:          Should be true
                Test:           TestPeople/unfollow/badger
                Messages:       did not find peer alice in graph lookup table (len:0)
level=warn t=32.77289ms msg="authbypass - trust on first use"
    people_test.go:274:
                Error Trace:    /home/go-ssb/go-ssb/graph/people_test.go:274
                Error:          Received unexpected error:
                                auth assert: host(alice) accepted remote(bob) (dist:0)
                Test:           TestPeople/unfollow/badger
                Messages:       assertion #3 failed
    people_test.go:278: warning: failed to dump graph to SVG RenderSVG: dot command failed: exec: "dot": executable file not found in $PATH
level=warn t=33.802143ms msg="authbypass - trust on first use"
    people_test.go:274:
                Error Trace:    /home/go-ssb/go-ssb/graph/people_test.go:274
                Error:          Received unexpected error:
                                follows assert failed - wanted true
                Test:           TestPeople/unfollow/badger
                Messages:       assertion #5 failed
    people_test.go:278: warning: failed to dump graph to SVG RenderSVG: dot command failed: exec: "dot": executable file not found in $PATH
2022/12/23 01:42:36 got err for badgerContacts error pouring read value of seq(0): error calling setter func: db/idx contacts: failed to update index. {Type: Contact:@rNM8Fp8TcXo7XXoFE3uFO/B7c+QdN+JdcPEc45VQlaI=.ed25519 Following:true Blocking:false}: failed to write big batch (1): error in badger transaction (update) 1: Writes are blocked, possibly due to DropAll or Close
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x89544b]

goroutine 131 [running]:
github.com/dgraph-io/badger/v3.(*memTable).IncrRef(...)
        /home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.5/memtable.go:231
github.com/dgraph-io/badger/v3.(*DB).getMemTables(0xc000153b00)
        /home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.5/db.go:696 +0x12b
github.com/dgraph-io/badger/v3.(*DB).get(0xc000153b00, {0xc0187bb830, 0x27, 0x27})
        /home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.5/db.go:730 +0x9d
github.com/dgraph-io/badger/v3.(*Txn).Get(0xc000204280, {0xc0000243a0, 0x1f, 0x20})
        /home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.5/txn.go:478 +0x2a7
github.com/ssbc/margaret/indexes/badger.(*index).GetSeq.func1(0xc000153b00?)
        /home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/indexes/badger/index.go:387 +0x32
github.com/dgraph-io/badger/v3.(*DB).View(0xc00ce7c8a0?, 0xc00cd80e10)
        /home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.5/txn.go:806 +0x9c
github.com/ssbc/margaret/indexes/badger.(*index).GetSeq(0xc0001ce5a0)
        /home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/indexes/badger/index.go:386 +0x198
github.com/ssbc/margaret/indexes.(*sinkIndex).QuerySpec(0x0?)
        /home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/indexes/sinkindex.go:30 +0x25
github.com/ssbc/go-ssb/graph.serveLog.func1()
        /home/go-ssb/go-ssb/graph/builder_test.go:330 +0xdd
created by github.com/ssbc/go-ssb/graph.serveLog
        /home/go-ssb/go-ssb/graph/builder_test.go:327 +0x165
FAIL    github.com/ssbc/go-ssb/graph    0.129s
FAIL
```

--- a/graph/builder.go
+++ b/graph/builder.go
@@ -84,6 +84,11 @@ func NewBuilder(log log.Logger, db *badger.DB, hmacSecret *[32]byte) *BadgerBuil
 
 		hmacSecret: hmacSecret,
 	}
+
+	// make sure we initialize the waitgroup so we have an opportunity to index
+	b.indexSyncStart()
+	defer b.indexSyncDone()
+
 	return b
 }
 

--- a/graph/builder_indexing.go
+++ b/graph/builder_indexing.go
@@ -152,6 +152,8 @@ func (b *BadgerBuilder) updateContacts(ctx context.Context, seq int64, val inter
 }
 
 func (b *BadgerBuilder) OpenContactsIndex() (librarian.SeqSetterIndex, librarian.SinkIndex) {
+	b.indexSyncStart()
+	defer b.indexSyncDone()
 	if b.idxSinkContacts == nil {
 		b.idxSinkContacts = librarian.NewSinkIndex(b.updateContacts, b.idx)
 	}

--- a/graph/builder_test.go
+++ b/graph/builder_test.go
@@ -32,6 +32,8 @@ import (
 	"github.com/ssbc/go-ssb/repo"
 )
 
+var indexesReady sync.WaitGroup
+
 func openUserMultilogs(t *testing.T) (multilog.MultiLog, multilog.Sink) {
 	r := require.New(t)
 
@@ -107,6 +109,8 @@ func makeBadger(t *testing.T) testStore {
 
 	_, announcementSink := builder.OpenAnnouncementIndex()
 	mfAnnounceErrc := serveLog(ctx, "badgerMetafeedAnnounce", tRootLog, announcementSink, true)
+
+	indexesReady.Wait()
 
 	tc.root = tRootLog
 	tc.gbuilder = builder
@@ -314,20 +318,43 @@ func (tc testStore) theScenario(t *testing.T) {
 }
 
 func serveLog(ctx context.Context, name string, l margaret.Log, snk librarian.SinkIndex, live bool) <-chan error {
+	indexesReady.Add(1)
 	errc := make(chan error)
 	go func() {
 		defer close(errc)
 
-		src, err := l.Query(snk.QuerySpec(), margaret.Live(live))
+		// first do a non-live query to process any messages which have already been posted
+		src, err := l.Query(snk.QuerySpec(), margaret.Live(false))
 		if err != nil {
-			log.Println("got err for", name, err)
+			log.Println("got non-live err for", name, err)
 			errc <- fmt.Errorf("%s query failed: %w", name, err)
 			return
 		}
 
 		err = luigi.Pump(ctx, snk, src)
 		if err != nil && !errors.Is(err, ssb.ErrShuttingDown) {
-			log.Println("got err for", name, err)
+			log.Println("got non-live err for", name, err)
+			errc <- fmt.Errorf("%s serve exited: %w", name, err)
+		}
+
+		if !live {
+			// not doing a live query, so we're done
+			indexesReady.Done()
+			return
+		}
+
+		// now do the live query
+		src, err = l.Query(snk.QuerySpec(), margaret.Live(live))
+		if err != nil {
+			log.Println("got live err for", name, err)
+			errc <- fmt.Errorf("%s query failed: %w", name, err)
+			return
+		}
+
+		indexesReady.Done()
+		err = luigi.Pump(ctx, snk, src)
+		if err != nil && !errors.Is(err, ssb.ErrShuttingDown) {
+			log.Println("got live err for", name, err)
 			errc <- fmt.Errorf("%s serve exited: %w", name, err)
 		}
 	}()

--- a/graph/people_test.go
+++ b/graph/people_test.go
@@ -19,6 +19,7 @@ import (
 	refs "github.com/ssbc/go-ssb-refs"
 	"github.com/ssbc/go-ssb/internal/storedrefs"
 	"github.com/ssbc/go-ssb/internal/testutils"
+	"github.com/ssbc/go-ssb/multilogs"
 )
 
 type PeopleOp interface {
@@ -260,6 +261,9 @@ func (tc PeopleTestCase) run(mk func(t *testing.T) testStore) func(t *testing.T)
 			err := op.Op(&state)
 			r.NoError(err, "error performing operation(%d) of %v type %T: %s", i, op, op)
 		}
+
+		// wait for the multilogs to catch up
+		multilogs.WaitUntilUserFeedIndexIsSynced()
 
 		// punch in nicks
 		g, err := state.store.gbuilder.Build()

--- a/graph/people_test.go
+++ b/graph/people_test.go
@@ -18,7 +18,6 @@ import (
 	"github.com/ssbc/go-ssb"
 	refs "github.com/ssbc/go-ssb-refs"
 	"github.com/ssbc/go-ssb/internal/storedrefs"
-	"github.com/ssbc/go-ssb/internal/testutils"
 	"github.com/ssbc/go-ssb/multilogs"
 )
 
@@ -243,11 +242,6 @@ type PeopleTestCase struct {
 
 func (tc PeopleTestCase) run(mk func(t *testing.T) testStore) func(t *testing.T) {
 	return func(t *testing.T) {
-		if testutils.SkipOnCI(t) {
-			// https://github.com/ssbc/go-ssb/issues/163
-			return
-		}
-
 		r := require.New(t)
 		a := assert.New(t)
 

--- a/multilogs/userfeeds.go
+++ b/multilogs/userfeeds.go
@@ -7,6 +7,8 @@ package multilogs
 import (
 	"context"
 	"fmt"
+	"sync"
+	"time"
 
 	refs "github.com/ssbc/go-ssb-refs"
 	"github.com/ssbc/go-ssb/internal/storedrefs"
@@ -16,7 +18,27 @@ import (
 
 const IndexNameFeeds = "userFeeds"
 
+var idxInSync sync.WaitGroup
+
+func indexSyncStart() {
+	idxInSync.Add(1)
+}
+
+func indexSyncDone() {
+	time.AfterFunc(100 * time.Millisecond, func() {
+		idxInSync.Done()
+	})
+}
+
+// WaitUntilUserFeedIndexIsSynced blocks until all the index processing is in sync with the rootlog
+func WaitUntilUserFeedIndexIsSynced() {
+	idxInSync.Wait()
+}
+
 func UserFeedsUpdate(ctx context.Context, seq int64, value interface{}, mlog multilog.MultiLog) error {
+	indexSyncStart()
+	defer indexSyncDone()
+
 	if nulled, ok := value.(error); ok {
 		if margaret.IsErrNulled(nulled) {
 			return nil

